src/EditorFeatures/CSharp.Wpf/Completion/CompletionProviders/CSharpReplCommandCompletionProvider.cs

```diff
@@ -74,10 +74,9 @@ internal override bool IsInsertionTrigger(SourceText text, int characterPosition
 
         internal override ImmutableHashSet<char> TriggerCharacters { get; } = CompletionUtilities.SpaceTriggerCharacter;
 
-        protected override async Task<bool> ShouldDisplayCommandCompletionsAsync(SyntaxTree tree, int position, CancellationToken cancellationToken)
-        {
-            return await tree.IsBeforeFirstTokenAsync(position, cancellationToken).ConfigureAwait(false) &&
-                   tree.IsPreProcessorKeywordContext(position, cancellationToken);
-        }
+        protected override Task<bool> ShouldDisplayCommandCompletionsAsync(SyntaxTree tree, int position, CancellationToken cancellationToken)
+            => Task.FromResult(
```

Comment:
⚠️  This needs to use cached return values, either manually from `SpecializedTasks` or automatically by using the state machine provided by `async`/`await`.

Reply:
I made the entire codepath here synchronous without tasks.